### ISSUE TITLE
no-node-access rule incorrectly triggering from RTK Query syntax in 7.5.4

### ISSUE THREAD

**Issue by MrZipper7** [testing-library/eslint-plugin-testing-library#1041]
### Have you read the Troubleshooting section?

Yes

### Plugin version

7.5.4

### ESLint version

v9.30.1

### Node.js version

20.19.3

### Bug description

After updating to 7.5.4, all instances of RTK Query's `endpoint.select()` method ([https://redux-toolkit.js.org/rtk-query/api/created-api/endpoints#select](https://redux-toolkit.js.org/rtk-query/api/created-api/endpoints#select)) are triggering `no-node-access` errors.

Ex:
```js
const data = api.endpoints.getAllPosts.select(undefined)(store.getState()).data?.data
                                       ~~~~~~
expect(data).toEqual(expectedData)
```

### Steps to reproduce

1. Define an RTK Query endpoint ([https://redux-toolkit.js.org/rtk-query/api/createApi#createapi-parameters](https://redux-toolkit.js.org/rtk-query/api/createApi#createapi-parameters).
2. Call the `select` function of the endpoint in a test.
3. Observe `the no-node-access` error.

### Error output/screenshots

<img width="1285" height="343" alt="Image" src="https://github.com/user-attachments/assets/49bd4bea-891d-4ceb-9dc7-53d5bbf24e9c" />

### ESLint configuration

```
  {
    files: ['src/**/__test__/*', 'src/**/*.test.ts', 'src/**/*.test.tsx'],
    extends: [
      testingLibrary.configs['flat/react'], 
      vitest.configs.recommended
    ],
    rules: {
      'testing-library/render-result-naming-convention': 'off',
    }
  },
```

### Rule(s) affected

`no-node-access`

### Do you want to submit a pull request to fix this bug?

No

**Comment by WikiRik** [comment]
I'm experiencing a similar issue I think. I'm accidentally running these rules on non-test files as well and I get a lint error on the following piece of code;
```
    const anchor = document.createElement("a");
    anchor.href = "/api/download?fileName=test.csv";
    anchor.click();
```

**Comment by spennine** [comment]
similar issue in our codebase:  element.click() is triggering the lint error for us as well

none of the files cited as violating the rule in our codebase are tests, they're all runtime code

```
                        const element = document.getElementById(id);
                        if (element && !checked) {
                          element.click();
                        }
```

**Comment by joshkel** [comment]
We're also seeing false positives with 7.5.4. In our case, it's from [Knex](https://knexjs.org/) database calls in our tests:

```ts
    const result = await db.select().table('DataRecord');
```

(We could probably try and change our setup to strictly separate tests that reference the DOM from tests that reference the database then change our ESLint rules to only run eslint-plugin-testing-library on DOM tests, but that's a decent-sized change.)

If this should be opened as a separate issue, please let me know.

**Comment by Belco90** [comment]
For those having issues in your codebase: [you should run this plugin only against test files](https://github.com/testing-library/eslint-plugin-testing-library#run-the-plugin-only-against-test-files).

**Comment by MrZipper7** [comment]
@Belco90 For clarification, at least in my case, this is happening in test files.

**Comment by denys-diedov** [comment]
We also have some problems with the new rules. https://testing-library.com/docs/react-testing-library/setup - We have a utils file, similar to this example.
import { renderWithProviders, screen, userEvent } from '@testUtils';
Or if renderWithProviders returns userEvent, ESLint will show an error. - ESLint: Avoid direct Node access. Prefer using the methods from Testing Library. (testing-library/no-node-access)
But initially, userEvent is an export directly from testing-library. Only in test files, of course.

**Comment by MrZipper7** [comment]
By way of further distinguishing from #1038, the files experiencing this issue are all unit tests that don't import from `testing-library` (or a custom export of `testing-library` utils) at all, and as far as I can tell, T-L doesn't appear anywhere up the import tree for them.

That makes me think that #1042 still might actually be useful here, but I'll be curious to see if #1043 ends up fixing it anyway as a side effect.

**Comment by Belco90** [comment]
Can you check if this is fixed in [v7.6.2](https://github.com/testing-library/eslint-plugin-testing-library/releases/tag/v7.6.2)?

**Comment by denys-diedov** [comment]
> Can you check if this is fixed in [v7.6.2](https://github.com/testing-library/eslint-plugin-testing-library/releases/tag/v7.6.2)?

Yes, for me it fixed the problem, updated, thanks!

**Comment by MrZipper7** [comment]
> Can you check if this is fixed in [v7.6.2](https://github.com/testing-library/eslint-plugin-testing-library/releases/tag/v7.6.2)?

@Belco90 - I'm not seeing any change on this issue after updating to 7.6.2

**Comment by Belco90** [comment]
> [@Belco90](https://github.com/Belco90) - I'm not seeing any change on this issue after updating to 7.6.2

Then changes from #1048 are needed

**Comment by MrZipper7** [comment]
> > [@Belco90](https://github.com/Belco90) - I'm not seeing any change on this issue after updating to 7.6.2
> 
> Then changes from [#1048](https://github.com/testing-library/eslint-plugin-testing-library/pull/1048) are needed

@Belco90 @y-hsgw -- reporting that #1048 and updating to 7.6.4 did not resolve this issue

**Comment by rzane** [comment]
Looks like there have been several attempts to fix this issue. I'm on 7.6.6. I'm getting a false positive on a test that looks like this:

```typescript
const subject = async (user: UserEvent) => {
  render(<SomeComponent />);
  await user.click(await screen.findByText('Continue'));
};
```

This is admittedly pretty unusual, so I don't know if it's worth trying to detect this scenario.
